src/vs/workbench/browser/parts/views/treeView.ts

```diff
@@ -348,6 +350,29 @@ abstract class AbstractTreeView extends Disposable implements ITreeView {
 		this._onDidChangeDescription.fire(this._description);
 	}
 
+	private _badge: IViewBadge | undefined;
+	private _badgeActivity: IDisposable | undefined;
+	get badge(): IViewBadge | undefined {
+		return this._badge;
+	}
+
+	set badge(badge: IViewBadge | undefined) {
+
+		if (this._badgeActivity) {
+			this._badgeActivity.dispose();
+			this._badgeActivity = undefined;
+		}
+
+		this._badge = badge;
+
+		if (badge) {
+			const activity = {
+				badge: new NumberBadge(badge.value, () => badge.tooltip)
+			};
```

Comment:
I am only setting the badge property of `IActivity` here, there are also `clazz` and `priority` properties, I couldn't determine if they were meaningful for this use or not.

Reply:
It looks like most VS Code core badges use the default `priority`. The progress service uses 100 for the priority. Let's set the priority to 150 so that the progress service wins. I think most people will want to see the progress more than the number badge.

Reply:
For `clazz` it looks like that lets us specify additional css classes to add to the html element that ends up being added for the badge. We don't need to add this for now.

Reply:
Done, I've updated the priority to `150` for both the tree and webviews.